src/Bridge/Graphql/Type/SchemaBuilder.php

```diff
@@ -0,0 +1,237 @@
+            ];
+        }
+
+        foreach ($this->getOperations($resourceMetadata, true, false) as $operationName => $queryCollectionOperation) {
+            $fieldNamePrefix = 'get' === $operationName ? '' : $operationName;
+            $queryFields[lcfirst($fieldNamePrefix.Inflector::pluralize($shortName))] = [
+                'type' => GraphQLType::listOf($resourceType),
+                'args' => [
+                    'first' => [
+                        'type' => GraphQLType::int(),
+                        'description' => 'Returns the first n elements from the list.',
+                    ],
+                ],
+                'resolve' => $this->collectionResolverFactory->createCollectionResolver($resource, $operationName),
+            ];
+        }
+
+        return $queryFields;
+    }
+
+    /**
+     * Gets the type of the given resource.
+     */
+    private function getResourceType(string $resource, ResourceMetadata $resourceMetadata): ObjectType
+    {
+        $shortName = $resourceMetadata->getShortName();
+
+        if (isset($this->resourceTypesCache[$shortName])) {
+            return $this->resourceTypesCache[$shortName];
+        }
+
+        return $this->resourceTypesCache[$shortName] = new ObjectType([
+            'name' => $shortName,
+            'description' => $resourceMetadata->getDescription(),
+            'fields' => function () use ($resource) {
+                return $this->getResourceTypeFields($resource);
+            },
+        ]);
+    }
+
+    /**
+     * Gets the fields of the type of the given resource.
+     */
+    private function getResourceTypeFields(string $resource): array
+    {
+        $fields = [];
+
+        foreach ($this->propertyNameCollectionFactory->create($resource) as $property) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resource, $property);
+
+            $propertyType = $propertyMetadata->getType();
+            if (null !== $propertyType) {
+                try {
+                    $fields[$property] = [
+                        'type' => $this->convertType($propertyType),
+                        'description' => $propertyMetadata->getDescription(),
+                    ];
+                } catch (InvalidTypeException $e) {
+                    continue;
+                }
+            }
+        }
+
+        return $fields;
+    }
+
+    /**
+     * Gets the field arguments of the identifier of a given resource.
+     */
+    private function getResourceIdentifierFieldArguments(string $resource): array
+    {
+        foreach ($this->propertyNameCollectionFactory->create($resource) as $property) {
+            $propertyMetadata = $this->propertyMetadataFactory->create($resource, $property);
+            $propertyIsIdentifier = $propertyMetadata->isIdentifier() ?? false;
+            $propertyType = $propertyMetadata->getType();
+
+            if (!$propertyIsIdentifier || null === $propertyType) {
```

Comment:
`|| !$propertyType`

Reply:
Sructinizr (legitimately) add warnings when not checking explicitly the type when dealing with `null`values. I would keep this as is.